scene/2d/audio_stream_player_2d.cpp

```diff
@@ -301,7 +301,8 @@ bool AudioStreamPlayer2D::is_playing() const {
 float AudioStreamPlayer2D::get_playback_position() {
 	// Return the playback position of the most recently started playback stream.
 	if (!stream_playbacks.is_empty()) {
-		return AudioServer::get_singleton()->get_playback_position(stream_playbacks[stream_playbacks.size() - 1]);
+		float time_since_last_mix = AudioServer::get_singleton()->get_time_since_last_mix() * pitch_scale;
```

Comment:
I think if we're subtracting this should be the time *until* the next mix. If we subtract the time since the last mix, the playback position will go down over the course of the mix, then jump up, go down again etc.

Reply:
This is true, but I tested again, and unfortunately negative numbers seem to happen. So to keep it safe it would probably be best adding instead of subtracting, unless the consensus is against it, and clamping the number is the right approach.

Reply:
I would expect negative numbers if you were to call this after calling play() and prior to the first mix completing so that's not too surprising. The bigger issue I see though is the pausing problem, and that the easiest way to solve the issue with pausing (an if statement) could cause the playback positions to not be monotonic.

Reply:
> easiest way to solve the issue with pausing (an if statement) could cause the playback positions to not be monotonic.

I have vigorously tested the simple fix for `stream_paused`. That seems to work surprisingly well.

However, a bit of an issue.
There are occasions with this PR where the next `playback position` value is, in fact, smaller than the previous value. But the difference between the two can be so negligible, it may be a floating-point rounding error.
![image](https://github.com/godotengine/godot/assets/66727710/90cac3a7-302d-465d-b5e0-4a8bace434ad)

Reply:
I agree that subtracting `get_time_to_next_mix` sounds the most sense reasonable.

Apart from that, I'm wondering if we should involve `AudioServer::get_output_latency()`? There's a difference between technical playback and audible playback.  
Possibly the best option would be to just document this in the func `get_playback_position` to avoid extra computation cost? And anybody who's interested can post-process it themselves.

Sadly, I can't comment on pausing. I only know that there's a fade out and I'm uncertain how exactly that affects `_last_mix_time`…

Reply:
@Mickeon: `get_time_to_next_mix` is sometimes negative, please see https://github.com/godotengine/godot/issues/49403. Maybe we really must apply latency…

Reply:
> Maybe we really must apply latency…

That sounds good on paper. At least, according to https://github.com/godotengine/godot/issues/49403's comments, that looks to be the solution.
But that's something outside of this PR. How to handle it here? Should we let `get_time_to_next_mix()` return a negative value for the playback_position formula? Or should it be clamped or otherwise modified, like `MAX(AudioServer::get_singleton()->get_time_to_next_mix() * pitch_scale, 0.0)`?

Reply:
Hm… I stand with my first comment: put latency info into the method docs. So in this case: "MAX" it as you suggest